language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/DefinedDataType.java

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data.codegen;
+
+import com.daml.ledger.javaapi.data.Value;
+
+public interface DefinedDataType<T> {
```

Comment:
Only an interface because `enum`s can only implement interfaces, not extend classes.